framework/include/materials/GenericConstantVectorMaterial.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+
+template <bool is_ad>
+class GenericConstantVectorMaterialTempl : public Material
+{
+public:
+  static InputParameters validParams();
+
+  GenericConstantVectorMaterialTempl(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void computeQpProperties() override;
+
+  std::string _prop_name;
+  const RealVectorValue & _prop_value;
```

Comment:
Seems like it might be useful to allow a user to create an arbitrary number of these vector properties like we do for scalar material properties? I could imagine a user trying to use this class the first time as they do with the scalar version and maybe being surprised that they can't

Reply:
_prop_values(getParam<std::vector<RealVectorValue>>("prop_values"))
only gets the first component of the vector if I try to define multiple vectors. 
Should I be adding a new template to getParam ?

Reply:
I ll just use a vector of Real and keep track of indices.

Reply:
Wait `getParam<std::vector<RealVectorValue>>()` doesn't return a `std::vector<RealVectorValue>`? If there's something wonky there, that's a bug we definitely need to fix. We shouldn't have to do any work-arounds.

Reply:
I think the issue lies with RealVectorValue. I was looking into why I need to regold. And my guess is that getParam<RealVectorValue>() already doesnt work

Reply:
That would be pretty crazy. We do `RealVectorValue` for things like gravity in Navier-Stokes and I've never noticed an issue. Seems like we would have gotten user reports if that didn't work. I could believe that `std::vector<RealVectorValue>` isn't working though because I don't think we that anywhere in the repository yet

Reply:
I don't want to make you spend a lot of time on this. You've got other things to do and what really matters is the PINSFV stuff. I'm totally fine if you decide to revert to the single prop design. We can add vector ability later in follow-on work if we want.

Reply:
Actually what you're doing with the vector of `Real` stuff would be a great stand-in if it works. Because the input-file syntax would not change if/when we do switch over to `std::vector<RealVectorValue>`

Reply:
Ok I ll post an issue and leave it at that